slasher/rpc/server.go

```diff
@@ -108,6 +113,8 @@ func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.Indexed
 func (ss *Server) IsSlashableBlock(ctx context.Context, req *ethpb.SignedBeaconBlockHeader) (*slashpb.ProposerSlashingResponse, error) {
 	ctx, span := trace.StartSpan(ctx, "detection.IsSlashableBlock")
 	defer span.End()
+	ss.proposeLock.Lock()
+	defer ss.proposeLock.Unlock()
 
```

Comment:
Do the locks have to be this early? Wouldn't it be better to have them under the nil checks?

Reply:
moved them down , thanks